src/wallet/receive.cpp

```diff
@@ -303,18 +303,23 @@ Balance GetBalance(const CWallet& wallet, const int min_depth, bool avoid_reuse)
             const bool is_trusted{CachedTxIsTrusted(wallet, wtx, trusted_parents)};
             const int tx_depth{wallet.GetTxDepthInMainChain(wtx)};
             const CAmount tx_credit_mine{CachedTxGetAvailableCredit(wallet, wtx, ISMINE_SPENDABLE | reuse_filter)};
+            const CAmount tx_credit_mine_used{
+                avoid_reuse ? CachedTxGetAvailableCredit(wallet, wtx, ISMINE_SPENDABLE | ISMINE_USED) : tx_credit_mine};
```

Comment:
What about caculating `avoid_reuse` locally (e.g. `const bool avoid = !(m_wallet->IsWalletFlagSet(WALLET_FLAG_AVOID_REUSE));` at the beginning of the function), instead of passing it? I thought @achow101's [comment](https://github.com/bitcoin/bitcoin/pull/28776/files#r1407875236) was referring to include that too but maybe I'm wrong.

Reply:
> have it always compute the used balance for us rather than having the caller do this extra computation.

My understanding of this part of @achow101  [comment](https://github.com/bitcoin/bitcoin/pull/28776/files#r1407875236) was to move the calculation for the `used` balance to `GetBalance(...)` 

I have updated the PR and added `GetFullBalance()` which adds `used` balance to `GetBalance(...)`